opm/simulators/flow/EclWriter.hpp

```diff
@@ -389,31 +389,47 @@ class EclWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::Grid>
 
     void writeReports(const SimulatorTimer& timer)
     {
-        auto rstep = timer.reportStepNum();
-
-        if ((rstep > 0) && (this->collectOnIORank_.isIORank())) {
-            const auto& rpt = this->schedule_[rstep-1].rpt_config.get();
-            if (rpt.contains("WELLS") && rpt.at("WELLS") > 0) {
-                outputModule_->outputTimeStamp("WELLS",
-                                               timer.simulationTimeElapsed(),
-                                               rstep,
-                                               timer.currentDateTime());
-                outputModule_->outputProdLog(rstep - 1, rpt.at("WELLS") > 1);
-                outputModule_->outputInjLog(rstep - 1, rpt.at("WELLS") > 1);
-                outputModule_->outputCumLog(rstep - 1, rpt.at("WELLS") > 1);
-                outputModule_->outputMSWLog(rstep - 1);
-            }
+        if (! this->collectOnIORank_.isIORank()) {
+            return;
+        }
 
-            outputModule_->outputFipAndResvLog(inplace_,
-                                               rstep,
-                                               timer.simulationTimeElapsed(),
-                                               timer.currentDateTime(),
-                                               false,
-                                               simulator_.gridView().comm());
+        // SimulatorTimer::reportStepNum() is the simulator's zero-based
+        // "episode index".  This is generally the index value needed to
+        // look up objects in the Schedule container.  That said, function
+        // writeReports() is invoked at the *beginning* of a report
+        // step/episode which means we typically need the objects from the
+        // *previous* report step/episode.  We therefore need special case
+        // handling for reportStepNum() == 0 in base runs and
+        // reportStepNum() <= restart step in restarted runs.
+        const auto firstStep = this->initialStep();
+        const auto simStep =
+            std::max(timer.reportStepNum() - 1, firstStep);
+
+        const auto& rpt = this->schedule_[simStep].rpt_config();
+
+        if (rpt.contains("WELSPECS") && (rpt.at("WELSPECS") > 0)) {
```

Comment:
maybe add a temp to add dual lookup

Reply:
> maybe add a temp to add dual lookup

Do you mean something like the typical pattern
```C++
auto pos = map.find(kw);
if (pos != map.end() && pos->second > 0) {
    // do something
}
```
If so, that's not (currently) possible.  The [`RPTConfig`](https://github.com/OPM/opm-common/blob/3d86149501a42e02daadafd5e29a995544c18527/opm/input/eclipse/Schedule/RPTConfig.hpp#L39) class does not (currently) have a `find()` member function so we're (currently) forced to use `contains()` + `at()` to look up elements.

I can add a `RPTConfig::find()` function if you'd like.

Reply:
> ```
> const bool has_welspec = rpt.contains("WELSPECS") && rpt.at("WELSPECS") > 0;
> ```

I don't understand what you mean.  The second lookup is for `"WELLS"`, not `"WELSPECS"`.

Reply:
doh. sorry for the noise.